### PR TITLE
PKCS12 etc.: Add hints on using -legacy and -provider-path options

### LINKED ISSUE #14790: Trouble reading legacy PKCS#12 files produced with OpenSSL 1.1.1
tested with 86522324d2a398ad3ed2f648a66b62b0a6176258 :
```
../openssl pkcs12 -in store.p12 -info
```
results in:
> Enter Import Password:
> MAC: sha1, Iteration 2048
> MAC length: 20, salt length: 8
> PKCS7 Encrypted data: pbeWithSHA1And40BitRC2-CBC, Iteration 2048
> Error outputting keys and certificates
> 80323002737F0000:error:0308010C:digital envelope > routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:330:Global default library context, Algorithm (RC2-40-CBC : 62), Properties ()

If *store.p12* was written with version 1.1.1f

The man page for the pkcs12 command recommends the   *-legacy* option for such problems.

But 
```
../openssl pkcs12 -legacy -in store.p12 -info
```
results in 
> openssl pkcs12: unable to load provider legacy

The legacy provider is not loaded by default.

### PR DESCRIPTION
**Pull request by DDvO**
Fixes #14790.

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man1/openssl-pkcs12.pod.in [hunk 1]
```diff
@@ -85,8 +85,12 @@ The PKCS#12 export encryption and MAC options such as B<-certpbe> and B<-iter>
 and many further options such as B<-chain> are relevant only with B<-export>.
 Conversely, the options regarding encryption of private keys when outputting
 PKCS#12 input are relevant only when the B<-export> option is not given.
+
 The default encryption algorithm is AES-256-CBC with PBKDF2 for key derivation.
 
+On trouble loading legacy PKCS#12 files involving for instance C<RC2-40-CBC>,
```
Comment (paulidale) [inline doc/man1/openssl-pkcs12.pod.in:91]: How about this instead:

> When encountering problems loading legacy PKCS#12 files that involve, for example, C<RC2-40-CBC>, try using the B<-legacy> option and, if needed, the B<-provider-path> option.

File: doc/man1/openssl-pkcs12.pod.in [hunk 2]
```diff
@@ -132,6 +136,10 @@ and so the input is just verified.
 =item B<-legacy>
 
 Use legacy mode of operation and automatically load the legacy provider.
+If OpenSSL is not installed system-wide, it is necessary to use in addition,
```
Comment (paulidale) [inline doc/man1/openssl-pkcs12.pod.in:139]: > it is necessary to also use, for example, B<-provider-path ./providers> or to set

**Review by paulidale — COMMENTED** [review]
A couple of wording suggestions.  Otherwise, this looks good.

File: doc/man1/openssl-pkcs12.pod.in (hunk 1 above)
Comment (DDvO) [reply doc/man1/openssl-pkcs12.pod.in]: Yes, sounds better - done.

File: doc/man1/openssl-pkcs12.pod.in (hunk 2 above)
Comment (DDvO) [reply doc/man1/openssl-pkcs12.pod.in]: Done

**Review by paulidale — APPROVED** [review]

File: doc/man1/openssl-pkcs12.pod.in [hunk 3]
```diff
@@ -132,6 +137,11 @@ and so the input is just verified.
 =item B<-legacy>
 
 Use legacy mode of operation and automatically load the legacy provider.
+If OpenSSL is not installed system-wide,
+it is necessary to also use, for example, B<-provider-path ./providers>
```
Comment (t8m) [inline doc/man1/openssl-pkcs12.pod.in:141]: Sorry for nitpicking: `C<-provider-path ./providers>`

File: doc/man1/openssl-pkcs12.pod.in [hunk 4]
```diff
@@ -85,8 +85,13 @@ The PKCS#12 export encryption and MAC options such as B<-certpbe> and B<-iter>
 and many further options such as B<-chain> are relevant only with B<-export>.
 Conversely, the options regarding encryption of private keys when outputting
 PKCS#12 input are relevant only when the B<-export> option is not given.
+
 The default encryption algorithm is AES-256-CBC with PBKDF2 for key derivation.
 
+When encountering problems loading legacy PKCS#12 files that involve,
+for example, C<RC2-40-CBC>,
```
Comment (t8m) [inline doc/man1/openssl-pkcs12.pod.in:92]: Not sure if `C<>` is right here

File: doc/man1/openssl-pkcs12.pod.in (hunk 3 above)
Comment (DDvO) [reply doc/man1/openssl-pkcs12.pod.in]: All right, changed.

File: doc/man1/openssl-pkcs12.pod.in (hunk 4 above)
Comment (DDvO) [reply doc/man1/openssl-pkcs12.pod.in]: Removed (in analogy to AES-256-CBC above).

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
This was quick - thanks and merged.

**Comment by DDvO** [comment]
Oops, last merge attempt failed for some reason.
Squashed and merged now successfully.

**Pull request closed without merging**